Simplify shell integration manual install by deferring to the code CLI

Currently install is a little cumbersome, for example:

> Add the following to your [PowerShell profile](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_profiles?view=powershell-7.2), replacing `<InstallDir>` with VS Code's installation directory. Run `code $Profile` in pwsh to open the file in VS Code.
>```pwsh
>if ($env:TERM_PROGRAM -eq "vscode") {
>  . "<InstallDir>\resources\app\out\vs\workbench\contrib\terminal\browser\media\shellIntegration.ps1"
>}
>```

It would be much nicer if this was encapsulated within the CLI, something like:

```
code shell-integration
```

We should also make sure these cases are covered:

- Remote development extensions - injection works currently but manual install doesn't talk about the server paths
- Regular SSH - it's not so easy to get the shell integration script to the remote machine

When inside Remote - WSL or Remote - SSH (probably Containers too), the Resources Dir can be found with:

```zsh
realpath "$(dirname "$(command -v code)")/../../resources"
```

- `command -v` vs `which`: https://www.shellcheck.net/wiki/SC2230

- calling `realpath` is better than `readlink -f` since it at least warns the user if the path can't be resolved.

- echo isn't required anymore

In such case, the latter snippet also needs to be adjusted:

```diff
if [ "$TERM_PROGRAM" == "vscode" ]; then
-   . <InstallDir>/app/out/vs/workbench/contrib/terminal/browser/media/shellIntegration-bash.sh
+   . <InstallDir>/out/vs/workbench/contrib/terminal/browser/media/shellIntegration-bash.sh
fi
```

And when not running through Remote, I wasn't able to find such a file in a Linux .deb installation. Even by following the original instructions, which would probably resolve to a path like: `/usr/bin/resources` which isn't valid too.

The correct path would be `/usr/share/code/resources` instead. And, in such case, the latter snippet would be correct.

--- a/src/vs/server/node/server.cli.ts
+++ b/src/vs/server/node/server.cli.ts
@@ -44,6 +44,7 @@ const isSupportedForCmd = (optionId: keyof RemoteParsedArgs) => {
 		case 'enable-smoke-test-driver':
 		case 'extensions-download-dir':
 		case 'builtin-extensions-dir':
+		case 'shell-integration':
 		case 'telemetry':
 			return false;
 		default:

